src/bridges/labwc/labwc_bridge.py

```diff
@@ -0,0 +1,737 @@
+        match key:
+            case "speed":
+                schema = "pointerSpeed"
+                textvalue = str(settings[key])
+            case "accel-profile":
+                schema = "accelProfile"
+                textvalue = 'adaptive' if settings[key] == "adaptive" else 'flat' # assume "default" is flat
+            case "tap-button-map":
+                schema = "tapButtonMap"
+                textvalue = settings[key]
+            case "click-method":
+                schema = "clickMethod"
+                if settings[key] == "none":
+                    textvalue = "none"
+                elif settings[key] == "areas":
+                    textvalue = "buttonAreas"
+                else:
+                    textvalue = "clickfinger"
+            case "send-events":
+                schema = "sendEventsMode"
+                if settings[key] == "enabled":
+                    textvalue = "yes"
+                elif settings[key] == "disabled":
+                    textvalue = "no"
+                else:
+                    textvalue = "disabledOnExternalMouse"
+            case "double-click":
+                textvalue = str(settings[key])
+            case _:
+                if key in yesno:
+                    schema = yesno[key]
+                    textvalue = 'yes' if settings[key] else 'no'
+                else:
+                    self.log.info("unknown key " + key + " for peripherals category " + category)
+                    return
+
+        if key == "double-click":
+            schema = "./mouse/doubleClickTime"
+        else:
+            schema = "./libinput/device[@category='" + category + "']/" + schema
+        bridge = self.et.find(schema)
+
+        if bridge is not None:
+            bridge.text = textvalue
+        else:
+            self.log.info("cannot find schema " + schema + " to set the value " + value)
+
+        self.write_config()
+
+    # this handles keyboard layout changes
+    def desktop_input_sources_changed(self, settings, key):
+        if key != "sources":
+            return
+
+        # grab the settings sources and reformat it
+        # i.e. variants expressed as "+variant" need to be
+        # converted to (variant)
+        # and we ignore ibus keyboard layouts since the
+        # window manager expects only xkb
+        layout = ""
+        for source in settings[key]:
+            if source[0] == 'xkb':
+                extract = source[1].replace("'","")
+
+                if "+" in extract:
+                    rhs = extract.split("+")
+                    extract = rhs[0] + "(" + rhs[1] + ")"
+
+                if layout == "":
+                    layout = extract
+                else:
+                    layout += "," + extract
+
+        if layout == "":
+            layout = "us" # default to at least a known keyboard layout
+
+        path = os.path.join(os.environ["HOME"], ".config", "labwc", "environment")
```

Comment:
I think these paths have been repeated enough that we should just have a function to get the labwc path to a file, e.g. `getLabwcPathToFile(filename)`